diff --git a/platform/qnx/detect.py b/platform/qnx/detect.py
index 597b2ac..2ae8dfd 100644
--- a/platform/qnx/detect.py
+++ b/platform/qnx/detect.py
@@ -32,7 +32,7 @@ def get_opts():
     return [
         EnumVariable("linker", "Linker program", "default", ("default", "bfd", "gold", "lld", "mold")),
         BoolVariable("use_llvm", "Use the LLVM compiler", False),
-        BoolVariable("use_static_cpp", "Link libgcc and libstdc++ statically for better portability", True),
+        BoolVariable("use_static_cpp", "Link libgcc and libstdc++ statically for better portability", False),
         BoolVariable("use_coverage", "Test Godot coverage", False),
         BoolVariable("use_ubsan", "Use LLVM/GCC compiler undefined behavior sanitizer (UBSAN)", False),
         BoolVariable("use_asan", "Use LLVM/GCC compiler address sanitizer (ASAN)", False),
@@ -82,13 +82,6 @@ def configure(env: "SConsEnvironment"):
     supported_arches = ["x86_32", "x86_64", "arm32", "arm64"]
     validate_arch(env["arch"], get_name(), supported_arches)
 
-    ## Build type
-
-    if env.dev_build:
-        # This is needed for our crash handler to work properly.
-        # gdb works fine without it though, so maybe our crash handler could too.
-        env.Append(LINKFLAGS=["-rdynamic"])
-
     # Cross-compilation
     qnx_host = get_env_qnx_host()
     qnx_target = get_env_qnx_target()
@@ -109,8 +102,9 @@ def configure(env: "SConsEnvironment"):
         cross_compile = "ntox86_64"
 
     ## Compiler configuration
-    env["CC"] = f"{compiler_path}/{cross_compile}-gcc"
-    env["CXX"] = f"{compiler_path}/{cross_compile}-g++"
+    if not env["use_llvm"]:
+        env["CC"] = f"{compiler_path}/{cross_compile}-gcc"
+        env["CXX"] = f"{compiler_path}/{cross_compile}-g++"
 
 
     if env["linker"] != "default":
@@ -208,8 +202,6 @@ def configure(env: "SConsEnvironment"):
             env["RANLIB"] = "gcc-ranlib"
             env["AR"] = "gcc-ar"
 
-    env.Append(CCFLAGS=["-pipe"])
-
     ## Dependencies
 
     if env["use_sowrap"]:
