diff --git a/platform/qnx/detect.py b/platform/qnx/detect.py
index d576ffe57d..40fc9e9baf 100644
--- a/platform/qnx/detect.py
+++ b/platform/qnx/detect.py
@@ -227,15 +227,20 @@ def configure(env: "SConsEnvironment"):
 
     if not env["builtin_freetype"]:
         env.ParseConfig("pkg-config freetype2 --cflags --libs")
+        env.Append(LIBS=["bz2"])
 
     if not env["builtin_graphite"]:
         env.ParseConfig("pkg-config graphite2 --cflags --libs")
 
     if not env["builtin_icu4c"]:
         env.ParseConfig("pkg-config icu-i18n icu-uc --cflags --libs")
+        env.Append(LIBS=["icudata"])
 
     if not env["builtin_harfbuzz"]:
         env.ParseConfig("pkg-config harfbuzz harfbuzz-icu --cflags --libs")
+        env.ParseConfig("pkg-config glib-2.0 --cflags --libs")
+        env.Append(LIBS=["iconv"])
+        env.ParseConfig("pkg-config libpcre2-8 --cflags --libs")
 
     if not env["builtin_icu4c"] or not env["builtin_harfbuzz"]:
         print_warning(
@@ -274,6 +279,7 @@ def configure(env: "SConsEnvironment"):
 
     if not env["builtin_libwebp"]:
         env.ParseConfig("pkg-config libwebp --cflags --libs")
+        env.Append(LIBS=["sharpyuv"])
 
     if not env["builtin_mbedtls"]:
         # mbedTLS only provides a pkgconfig file since 3.6.0, but we still support 2.28.x,
@@ -323,13 +329,13 @@ def configure(env: "SConsEnvironment"):
             if os.system("pkg-config --exists alsa") == 0:  # 0 means found
                 env.ParseConfig("pkg-config alsa --cflags --libs")
                 env.Append(CPPDEFINES=["ALSA_ENABLED"])
-                env.Append(LIBS=["asound"])
+                env.Append(LIBS=["asound","slog2"])
             else:
                 print_warning("ALSA development libraries not found. Disabling the ALSA audio driver.")
                 env["alsa"] = False
         else:
             env.Append(CPPDEFINES=["ALSA_ENABLED"])
-            env.Append(LIBS=["asound"])
+            env.Append(LIBS=["asound","slog2"])
 
     if env["pulseaudio"]:
         if not env["use_sowrap"]:
